### PR TITLE
feat: add remove_view

### LINKED ISSUE #425: Implement remove_view in Client
### Summary

Implement remove_view that reverse of add_view in Client.

### What is the feature request for?

The core library

### The Problem

Client class has method `add_view`.
This is useful, but if you do it in the constructor of the extension's cog, it will cause a memory leak unless you delete the registered view when the extension is reloaded.
However, `remove_view`, which is the opposite of `add_view` to remove a view, is not implemented.
This is very inconvenient.
Not only does this cause a memory leak, but if you pass an instance of the cog when instantiating the view, the old instance will remain and will be used.
This means that the old version of the cog will be used in the View.
So I thought it would be good to have `remove_view`.

### The Ideal Solution

Example on cog
```python
class ExampleCog(commands.Cog):
  def __init__(self, bot):
    self.view = MyView()
    bot.add_view(self.view)

  def cog_unload(self):
    bot.remove_view(self.view)
```

### The Current Solution

It's a little aggressive, but that's how I'm dealing with it now.
```python
# Remove view added
for key, value in list(bot._connection._view_store._views.items()):
  if key[2] == "CustomID on the view":
    del bot._connection._view_store._views[key]
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by MaskDuck**
## Summary

close #425 

## Checklist

- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
  - [x] I have run `task pyright` and fixed the relevant issues.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Skelmis** [comment]
Fix your description.

File: nextcord/client.py [hunk 1]
```diff
@@ -1818,6 +1818,34 @@ def add_view(self, view: View, *, message_id: Optional[int] = None) -> None:
 
         self._connection.store_view(view, message_id)
 
+    def remove_view(self, view: View, *, message_id: Optional[int] = None) -> None:
```
Comment (Skelmis) [inline nextcord/client.py:1821]: `message_id` isn't used? Is it meant to be or was this a typo

Comment (MaskDuck) [reply nextcord/client.py:1821]:
```suggestion
    def remove_view(self, view: View) -> None:
```

File: nextcord/client.py [hunk 2]
```diff
@@ -1818,6 +1818,34 @@ def add_view(self, view: View, *, message_id: Optional[int] = None) -> None:
 
         self._connection.store_view(view, message_id)
 
+    def remove_view(self, view: View) -> None:
+        """Remove a :class:`nextcord.ui.View` for persistant listening.
```
Comment (DenverCoder1) [inline nextcord/client.py:1822]:
```suggestion
        """Remove a :class:`nextcord.ui.View` for persistent listening.
```

File: nextcord/client.py [hunk 3]
```diff
@@ -1818,6 +1818,34 @@ def add_view(self, view: View, *, message_id: Optional[int] = None) -> None:
...
+
+        .. versionadded:: 2.0
+
+        Parameters
+        ------------
+        view: :class:`nextcord.ui.View`
+            The view to register for dispatching.
```
Comment (DenverCoder1) [inline nextcord/client.py:1829]:
```suggestion
            The view to remove from dispatching.
```

File: nextcord/client.py [hunk 4]
```diff
@@ -1818,6 +1818,34 @@ def add_view(self, view: View, *, message_id: Optional[int] = None) -> None:
 
         self._connection.store_view(view, message_id)
 
+    def remove_view(self, view: View) -> None:
+        """Remove a :class:`nextcord.ui.View` from persistent listening.
+
+        .. versionadded:: 2.0
+
+        Parameters
+        ------------
```
Comment (ooliver1) [inline nextcord/client.py:1827]:
```suggestion
        Parameters
        ----------
```

File: nextcord/client.py [hunk 5]
```diff
@@ -1818,6 +1818,34 @@ def add_view(self, view: View, *, message_id: Optional[int] = None) -> None:
...
+        view: :class:`nextcord.ui.View`
+            The view to remove from dispatching.
+
+        Raises
+        -------
```
Comment (ooliver1) [inline nextcord/client.py:1832]:
```suggestion
        Raises
        ------
```

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Review by ooliver1 — CHANGES_REQUESTED** [review]
has this been tested? please test PRs before submitting for review
```py
@bot.event
async def on_ready():
    bot.add_view(MyView(timeout=None))
    print(bot._connection._view_store._views)
    bot.remove_view(MyView(timeout=None))
    print(bot._connection._view_store._views)
```
prints the same value both times

`nextcord.state.ConnectionState.remove_view` does not add a `message_id` to the tuple, so it will never pop anything as a tuple of 2 items will never match tuples of 3

**Comment by ooliver1** [comment]
Do you not think users should be able to customise that message_id variable to remove views made using that?

**Review by Skelmis — CHANGES_REQUESTED** [review]
To add to Oliver's comment with a bit more of a code touch, I would expect the tests found [here](https://nexttest.koldfusion.xyz/nextcord/706) to pass before approving this pr

**Comment by ooliver1** [comment]
Because of https://github.com/nextcord/nextcord/pull/682#issuecomment-1183046000, you were no longer able to make contributions to nextcord for the time being, so this PR is being closed as it would go stale from no updates.

**Pull request closed without merging**